dio_cache_interceptor/lib/src/dio_cache_interceptor.dart

```diff
@@ -30,6 +30,16 @@ class DioCacheInterceptor extends Interceptor {
 
     final cacheOptions = _getCacheOptions(options);
 
+    if (cacheOptions.callResponseInterceptorsAfterNotModified) {
```

Comment:
Why do you need this? Can we get rid of `callResponseInterceptorsAfterNotModified`?
If RequestOptions has a custom `validateStatus` which allows 304 as valid there should be nothing to do here.

Reply:
I added this because using validateStatus to validate 304's is not transparent to me. You would have to look into the source code of this cache interceptor to know how 304's are handled. Ofcourse this is more an issue with dio (handling not valid responses in onError and never going back to the onResponse flow) and it breaks every other response interceptor.